examples/fair_prep_results.py

```diff
@@ -0,0 +1,141 @@
+                                fnr = line.split(',')
+                                fnr_val = fnr[-1].rstrip("\n")
+                                present_fnr.append(float(fnr_val))
+                            
+                            elif line.startswith("test") and 'generalized_true_positive_rate' in line and 'None' in line:
+                                fpr = line.split(',')
+                                fpr_val = fpr[-1].rstrip("\n")
+                                present_fpr.append(float(fpr_val))
+                shutil.rmtree(path+'/'+subfolder)            
+    except:
+        pass
+    
+    return present_acc,present_di,present_fnr,present_fpr
+
+def exp_run_pre_pro_param(seed,learn,pre,post):
+    '''
+    Experiment function to run the experiments
+    '''
+    exp = GermanCreditDatasetSexExperiment(
+    fixed_random_seed=seed,
+    train_data_sampler=CompleteData(),
+    missing_value_handler=CompleteCaseAnalysis(),
+    numeric_attribute_scaler=NamedStandardScaler(),
+    learners=[learn],
+    pre_processors=[pre],
+    post_processors=[post]
+    )
+    
+    exp.run()
+
+#Mergining processors to one format for input
+processors = [[NoPreProcessing(),NoPostProcessing()], [DIRemover(1.0),NoPostProcessing()], [DIRemover(0.5),NoPostProcessing()], [Reweighing(),NoPostProcessing()],
+              [NoPreProcessing(), RejectOptionPostProcessing()],[NoPreProcessing(),CalibratedEqualOddsPostProcessing()]]
+
+
+
+accuracy, disp_imp, fpr, fnr  = [], [], [], []
+
+def run_exp(accuracy,disp_imp,fnr,fpr,seeds,learners,processors):
+	'''
+	This is the main driver function that calls the exp_run to run on combinations of various learners, pre and post processing techniques
+	'''
+	for processor in processors:
+	    for learn in learners:          
+	    	present_acc, present_di, present_fnr, present_fpr = [], [], [], []            
+	    	for seed in seeds:           
+	    		exp_run_pre_pro_param(seed, learn, pre = processor[0], post = processor[1])                
+	    		present_acc, present_di, present_fnr, present_fpr = extract_info(present_acc,present_di, present_fnr, present_fpr)            
+	    	acuracy.append(present_acc)            
+	    	disp_imp.append(present_di)
+	    	fnr.append(present_fnr)            
+	    	fpr.append(present_fpr)
+
+	return accuracy, disp_imp, fnr, fpr    
+
+accuracy, disp_imp, fnr, fpr  = run_exp(accuracy,disp_imp,fnr,fpr,seeds,learners,processors)
+
+def plotter(p_x, p_y, x_ticks, xlabel, main_title):
+    '''
+    Function to plot various comparison plots.
+    '''
+    title_list = ['NoPreProcessing','DIRemover(1.0)','DIRemover(0.5)','Reweighing','Reject Option','Caliberated Equal Odds']
+    label_list = [['NonTunedLogistic','TunedLogistic'],['NonTunedDecisionTree','TunedDecisionTree']] 
+    fig, axs = plt.subplots(6,2,figsize=((10,20)))
+    axs = axs.flatten()
+    for i in range(0,len(accuracy),2):
+        axs[i//2].scatter(p_x[i], accuracy[i],c='b',marker='o')
```

Comment:
the computation of i//2 is repeated many times, just compute it once

Reply:
changed